Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -815,24 +817,20 @@ public DensityOutput getSimpleDensity(double[] point) {
      *                              missing values.
      * @return A point with the missing values imputed.
      */
-    public double[] imputeMissingValues(double[] point, int numberOfMissingValues, int[] missingIndexes) {
-        checkArgument(numberOfMissingValues >= 0, "numberOfMissingValues must be greater than or equal to 0");
+    public ArrayList<?> getSimpleConditionalField(double[] point, int numberOfMissingValues, int[] missingIndexes) {
```

Comment:
I think it's a little ugly to have `ArrayList<?>` as the return type in this public API, requiring the user to cast the result in order to use it. We could have this method return `ArrayList<double[]>`, where the arrays would be single-element arrays when number of missing values is 1.

Reply:
done